fix(split/stack): allow pf-m-fill to apply to any direct child

mcoker opened the pull request:
fixes https://github.com/patternfly/patternfly/issues/7190

mcoker commented on src/patternfly/layouts/Split/examples/Split.md:
```diff
@@ -87,13 +87,13 @@ import './Split.css'
 ## Documentation
 ### Overview
-The split layout is designed to position items horizontally, with one item filling the available horizontal space.
+The split layout is designed to position items horizontally.
```
AFAIK there is no expectation that any (or just one) item(s) need to fill the space.

mcoker commented on src/patternfly/layouts/Split/examples/Split.md:
```diff
@@ -87,13 +87,13 @@ import './Split.css'
 | -- | -- | -- |
-| `.pf-v6-l-split` | `<div>`, `<section>`, or `<article>` | Initiates the split layout. |
-| `.pf-v6-l-split__item` | `<div>` | Initiates a split item. **Required** |
+| `.pf-v6-l-split` | `*` | Initiates the split layout. |
```
This can be any element, `<Split />` in pf-react allows a `component` prop to let you use any element here.

patternfly-build commented:
Preview: https://patternfly-pr-7201.surge.sh

mcoker commented on src/patternfly/layouts/Split/examples/Split.md:
```diff
@@ -87,13 +87,13 @@ import './Split.css'
-| `.pf-v6-l-split` | `<div>`, `<section>`, or `<article>` | Initiates the split layout. |
-| `.pf-v6-l-split__item` | `<div>` | Initiates a split item. **Required** |
+| `.pf-v6-l-split` | `*` | Initiates the split layout. |
+| `.pf-v6-l-split__item` | `*` | Initiates a split item. |
```
This is not required - prior to this PR, it was only required for `.pf-m-fill`, but with this PR it isn't required at all.

mcoker commented on src/patternfly/layouts/Split/split.scss:
```diff
@@ -12,10 +12,11 @@
-.#{$split}__item.pf-m-fill {
-  flex-grow: 1;
+  @at-root .#{$split}__item.pf-m-fill,
+  & > .pf-m-fill {
```
Now this rule renders as

```
.pf-v6-l-stack__item.pf-m-fill,
.pf-v6-l-stack > .pf-m-fill {
    flex-grow: 1;
}
```

andrewballantyne commented:
Is there a react rendering of this that I can test out my expectations @mcoker ?

mcoker commented:
@andrewballantyne sure! Here's a basic [`<Split />` codesandbox](https://codesandbox.io/p/sandbox/competent-keller-393jrh?file=%2Findex.js%3A12%2C19) and [`<Stack />` codesandbox](https://codesandbox.io/p/sandbox/heuristic-chatterjee-stpd99?file=%2Findex.js%3A15%2C38) with
* the CSS from this PR imported as `pf-item-update.css`
* updated the examples a little to show passing other children to the layout
* using `isFilled` on an item
* mimicking `isFilled` on a `<div>` (passing `className="pf-m-fill"`)

andrewballantyne commented:
Looks good to me @mcoker -- that's what I was looking for. I can't edit the sandboxes (don't have an account) but visually inspecting the examples looks to fit what I was after. Thanks!

srambach reviewed (approved):
Looks good - one question, would you want to adjust the custom workspace styling (that adds the dashed outline) to also work without the _item classes?
Also, is this change useful on the level layout?

mcoker commented:
@srambach sure I don't see why not. Added `.pf-m-fill` to level - just to confirm, was that what you were asking? We'll need a react follow up to add a prop for that. Also updated the level examples so the boxes are different heights since one of the things the level does is vertically center its items.

srambach reviewed (commented):
I think in discussions we decided we don't need to add a `pf-m-fill` on the level layout, but keep the local workspace styling to show that `__item` isn't required.

mcoker commented:
@srambach backed out the level layout's `.pf-m-fill` change https://github.com/patternfly/patternfly/pull/7201/commits/a2bfe807f8b21d1d92c0373e665fcfa61f6f59d6

patternfly-build commented:
:tada: This PR is included in version 6.1.0-prerelease.4 :tada:

The release is available on:
- [GitHub release](https://github.com/patternfly/patternfly/releases/tag/prerelease-v6.1.0-prerelease.4)
- [npm package (@prerelease dist-tag)](https://www.npmjs.com/package/@patternfly/patternfly/v/6.1.0-prerelease.4)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
